src/MerchantCenter/ShoppingContentDateTrait.php

```diff
@@ -0,0 +1,28 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent\Date as ShoppingContentDate;
+use Datetime;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Trait ShoppingContentDateTrait
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter
+ */
+trait ShoppingContentDateTrait {
+
+	/**
+	 * Convert ShoppingContentDate to DateTime.
+	 *
+	 * @param ShoppingContentDate $date The Google date.
+	 *
+	 * @return DateTime|false The date converted or false if the date is invalid.
+	 */
+	protected function convert_shopping_content_date( ShoppingContentDate $date ) {
+		return DateTime::createFromFormat( 'Y-m-d|', "{$date->getYear()}-{$date->getMonth()}-{$date->getDay()}" );
```

Comment:
Doesn't seem like it would be a bother in converting the date but is `|` a typo, or does it have a specific purpose?

Reply:
The `|` resets the time to 00:00:00, see this comment: https://github.com/php/doc-en/issues/671#issuecomment-1692909428.

I added it because each time I called this function, it would return the time depending on when it was executed.

Reply:
Nice, I learned something new. To be honest I haven't used createFromFormat before, but seems like a good use case here. My solution would have been to just use the [regular constructor](https://www.php.net/manual/en/datetime.construct.php) and then blanking out time values. Something like this:
```suggestion
		return new DateTime( "{$date->getYear()}-{$date->getMonth()}-{$date->getDay()}T00:00:00+00:00" );
```